src/theory/bags/bags_utils.cpp

```diff
@@ -683,11 +708,42 @@ Node NormalForm::evaluateBagMap(TNode n)
     ++it;
   }
   TypeNode t = nm->mkBagType(n[0].getType().getRangeType());
-  Node ret = NormalForm::constructConstantBagFromElements(t, mappedElements);
+  Node ret = BagsUtils::constructConstantBagFromElements(t, mappedElements);
+  return ret;
+}
+
+Node BagsUtils::evaluateBagFilter(TNode n)
+{
+  Assert(n.getKind() == BAG_FILTER);
+
+  // (bag.filter p (as bag.empty (Bag T)) = (as bag.empty (Bag T))
+  // (bag.filter p (bag "a" 3) ((bag "b" 2))) =
```

Comment:
This line doesnt parse, do you mean a disjoint union here?

Reply:
Yes, fixed.